server/tracedb/tracedb.go

```diff
@@ -21,20 +20,23 @@ const (
 )
 
 type TraceDB interface {
-	GetTraceByID(ctx context.Context, traceID string) (traces.Trace, error)
+	GetTraceByID(ctx context.Context, traceID string) (model.Trace, error)
 	Close() error
 }
 
 var ErrInvalidTraceDBProvider = fmt.Errorf("invalid traceDB provider: available options are (jaeger, tempo)")
 
 func New(c config.Config, repository model.RunRepository) (db TraceDB, err error) {
 	selectedDataStore, err := c.DataStore()
+
+	if selectedDataStore == nil && err == nil {
+		return nil, nil
```

Comment:
Returning nil here will force the rest of the code to do nil checks before accessing the datastore. Returning a `noopDataStore` is a safer approach. You can copy one from here: https://github.com/kubeshop/tracetest/blob/c78b2737da070ca529d2bd5f6443bae64f60719a/server/tracedb/tracedb.go#L28-L33